src/query/functions/src/scalars/map.rs

```diff
@@ -156,4 +158,112 @@ pub fn register(registry: &mut FunctionRegistry) {
             |map, _| map.values
         ),
     );
+
+    registry.register_2_arg_core::<NullableType<EmptyMapType>, NullableType<EmptyMapType>, EmptyMapType, _, _>(
+        "map_cat",
+        |_, _, _| FunctionDomain::Full,
+        |_, _, _| Value::Scalar(()),
+    );
+
+    registry.register_passthrough_nullable_2_arg(
+        "map_cat",
+        |_, domain1, domain2| {
+            FunctionDomain::Domain(match (domain1, domain2) {
+                (Some((key_domain1, val_domain1)), Some((key_domain2, val_domain2))) => Some((
+                    key_domain1.merge(key_domain2),
+                    val_domain1.merge(val_domain2),
+                )),
+                (Some(domain1), None) => Some(domain1).cloned(),
+                (None, Some(domain2)) => Some(domain2).cloned(),
+                (None, None) => None,
+            })
+        },
+        vectorize_with_builder_2_arg::<
+            MapType<GenericType<0>, GenericType<1>>,
+            MapType<GenericType<0>, GenericType<1>>,
+            MapType<GenericType<0>, GenericType<1>>,
+        >(|lhs, rhs, output_map, _| {
```

Comment:
```suggestion
        >(|lhs, rhs, output_map, ctx| {
```
and `ctx` params here, we need use `ctx.generics` to create the builder.

Reply:
No worries, it's a temp fix. We can skip it.